src/backend/cdb/motion/test/cdbsenddummypacket_test.c

```diff
@@ -269,6 +273,36 @@ test_send_dummy_packet_ipv6_to_ipv4(void **state)
 }
 
 
+static void
+test_send_dummy_packet_ipv4_to_ipv4_wildcard(void **state)
+{
+	break_loop = false;
+	int listenerSocketFd;
+	uint16 listenerPort;
+	int txFamily;
+
+	interconnect_address = NULL;
+	Gp_interconnect_address_type = INTERCONNECT_ADDRESS_TYPE_WILDCARD;
+	setupUDPListeningSocket(&listenerSocketFd, &listenerPort, &txFamily, &udp_dummy_packet_sockaddr);
+
+	Gp_listener_port = (listenerPort << 16);
+	UDP_listenerFd = listenerSocketFd;
+
+	ICSenderSocket = create_sender_socket(AF_INET);
+	ICSenderFamily = AF_INET;
+
+	SendDummyPacket();
+
+	const struct sockaddr_in *in = (const struct sockaddr_in *) &udp_dummy_packet_sockaddr;
+	assert_true(txFamily == AF_INET);
+	assert_true(in->sin_family == AF_INET);
+	assert_true(listenerPort == ntohs(in->sin_port));
+	assert_true(strcmp("0.0.0.0", inet_ntoa(in->sin_addr)) == 0);
+
+	wait_for_receiver(false);
+}
+
+
 static void
 test_send_dummy_packet_ipv6_to_ipv6_wildcard(void **state)
```

Comment:
the function is called `test_send_dummy_packet_ipv6_to_ipv6_wildcard`, but the test sets
`Gp_interconnect_address_type` to `INTERCONNECT_ADDRESS_TYPE_UNICAST`.

Reply:
Generally, this patch didn't use to touch existing test cases and only make default value explicit. The default value of `Gp_interconnect_address_type` is unicast so this "wildcard" test tested something on the unicast configuration. But as far as this test fully used to duplicate by `test_send_dummy_packet_ipv6_to_ipv6`, I took in account your notice  and replaced it with wildcard one. On Linux machines getaddrinfo returns ipv4 as a first item for wildcard request even on machines with ipv6 support, so this test is identical to ipv6_to_ipv4 now, but pin this implicit Linux behavior.

Reply:
Maybe we should rename the test to `test_send_dummy_packet_ipv6_to_ipv4` then? This seems confusing

Reply:
Why not test_send_dummy_packet_ipv4_to_ipv4? As far as receiving and sending sockets will use ipv4 despite of presence IPv6 in case of wildcard.

In my opinion, we pin fallback to IPv4 on Linux by this test.

Reply:
Isn't the sender socket explicitly AF_INET6? I think wildcard only affects the receiver socket in this test. And sorry, I meant `test_send_dummy_packet_ipv6_to_ipv6_wildcard` -> `test_send_dummy_packet_ipv6_to_ipv4_wildcard`

Reply:
@silent-observer, my bad, indeed we create the sender socket with hardcoded socket family in opposite to caller code of GPDB. GPDB use the same function to init both sender and receiver sockets, so socket families have to match always.

Finally, I've removed receiver socket types from names of wildcard tests, as far as we can't affect it. Is it ok for you?

Reply:
Yes, this makes sense